src/os.zig

```diff
@@ -0,0 +1,281 @@
+                            // calling process.
+                            //
+                            // Perhaps trying again later will work?
+                            return error.TryAgainLater;
+                        },
+                        .NFILE => {
+                            // The maximum number of files was open already in the system.
+                            //
+                            // Perhaps trying again later will work?
+                            return error.TryAgainLater;
+                        },
+                        .NOMEM, .RANGE => {
+                            // ENOMEM -- Insufficient memory to allocate passwd structure.
+                            // ERANGE -- Insufficient buffer space supplied.
+                            buffer_size *= buf_size_gain;
+                            temp_buf = try alloc.realloc(temp_buf, buffer_size);
+                            continue;
+                        },
+                        else => {
+                            // 0 or ENOENT or ESRCH or EBADF or EPERM or ... The given name or uid
+                            // was not found -- there's really no point in trying again.
+                            break;
+                        },
+                    }
+                }
+
+                return error.FailedToFindHomeDir;
+            }
+        };
+
+        if (bun.Environment.isWindows) {
+            @compileError("You cannot call queryPosix on Windows");
+        }
+
+        if (opts.user != null) {
+            // There is actually a reasonable desire to be able to query other users' home
+            // directories. However, we don't have a need to do that. See
+            // doi:10.1109/IEEESTD.2018.8277153 2.2.6.1 for further details.
+            @panic("Not implemented");
+        }
+
+        var result = fromEnv(bun.EnvVar.home.get());
+        if (result != error.FailedToFindHomeDir) {
+            return result;
+        }
+
+        result = Heuristics.getpwuidr(allocator);
+        if (result != error.FailedToFindHomeDir) {
+            return result;
+        }
+
+        return error.FailedToFindHomeDir;
+    }
+
+    /// Deduces the current user's home directory on Windows systems.
+    ///
+    /// Exactly matches the behavior of uv_os_homedir.
+    fn queryWin(allocator: std.mem.Allocator, opts: QueryOpts) QueryError!Self {
+        _ = allocator;
+        const Heuristics = struct {
+            pub fn getUserProfileDirectoryW() QueryError!Self {
+                const win32 = bun.windows;
+
+                var proc_token: win32.HANDLE = undefined;
+                defer _ = win32.CloseHandle(proc_token);
+                const proc_hndl = win32.GetCurrentProcess();
+                var rc = win32.OpenProcessToken(proc_hndl, win32.TOKEN_QUERY, &proc_token);
+                if (rc == win32.FALSE) {
+                    return error.FailedToFindHomeDir;
+                }
+
```

Comment:
```suggestion
                var proc_token: win32.HANDLE = undefined;
                const proc_hndl = win32.GetCurrentProcess();
                var rc = win32.OpenProcessToken(proc_hndl, win32.TOKEN_QUERY, &proc_token);
                if (rc == win32.FALSE) {
                    return error.FailedToFindHomeDir;
                }
                defer _ = win32.CloseHandle(proc_token);

```

the handle shouldn't be closed if OpenProcessToken fails